### PR TITLE
feat: add `run` keyword and support it for `run-with`

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze @JonathanStarup This PR adds the `run` keyword and supports it in `run-with`.

It was more painful than I thought, but perhaps the pain is acceptable?

Biggest loss for users will be having to escape `run` in `Runnable`. Second biggest loss -- like for `solve` -- will be that you cannot name a method `run`.

If we merge this, the next steps should be:
- We can split try-catch and run-with (which is nice I suppose?)
- We can split these code paths in Parser2
- We might want to use something other than the Indian Rupee for escape...

File: main/src/library/Random.flix [hunk 1]
```diff
@@ -77,7 +77,7 @@ mod Random {
     ///
     /// In other words, re-interprets the `Random` effect using the `IO` effect.
     ///
-    pub def run(f: Unit -> a \ ef): a \ (ef - Random) + IO = handle(f)()
+    pub def runWithHandler(f: Unit -> a \ ef): a \ (ef - Random) + IO = handle(f)()
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/library/Random.flix:80]: Maybe we just need `App.runAll`?

[the remark this answers is not in the document]
Comment (mlutze) [reply main/src/library/Random.flix:80]: > they don't compose

:O

**Comment by magnus-madsen** [comment]
@mlutze @JonathanStarup Do we still want to do this?

**Comment by mlutze** [comment]
> @mlutze @JonathanStarup Do we still want to do this?

Weak accept

File: main/src/library/Random.flix (hunk 1 above)
[the remark this answers is not in the document]
Comment (JonathanStarup) [reply main/src/library/Random.flix:80]: It's not useless, its good simple cases. `handle(..)()` is a bit more awkward than `run(..)`

**Comment by JonathanStarup** [comment]
I want something better but can't think of anything

**Comment by JonathanStarup** [comment]
It could be `{ .. } with Eff { .. }`

File: main/src/ca/uwaterloo/flix/language/phase/Lexer.scala [hunk 2]
```diff
@@ -345,8 +343,8 @@ object Lexer {
           TokenKind.Dot
         }
       case '$' if peek().isUpper => acceptBuiltIn()
-      case '₹' =>
-        // Don't include the rupee sign in the name
+      case '$' if peek().isLower =>
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Lexer.scala:346]: Does this break if a Java name has a `$` in it?

**Review by mlutze — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/Lexer.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Lexer.scala:346]: Thats my question: https://github.com/flix/flix/issues/9213

**Pull request merged**